sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample07_MockEventProcessorHandlersTests.cs

```diff
@@ -0,0 +1,244 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Processor;
+using Azure.Storage.Blobs;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests.Snippets
+{
+    /// <summary>
+    ///   The suite of tests defining the snippets used in the Event Hubs
+    ///   Sample07_MockEventProcessorHandlersTests sample.
+    /// </summary>
+    ///
+    [TestFixture]
+    [Category(TestCategory.Live)]
+    [Category(TestCategory.DisallowVisualStudioLiveUnitTesting)]
+    public class Sample07_MockEventProcessorHandlersTests
+    {
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProcessEventHandlerCheckpoint()
+        {
+            #region Snippet:EventHubs_Processor_Sample07_ProcessEventHandlerCheckpoint
+
+            // Application handler checkpoints after processing defined number of events.
+
+            const int EventsBeforeCheckpoint = 2;
+            var partitionEventCount = new ConcurrentDictionary<string, int>();
```

Comment:
How do you feel about simplifying this and assuming that we're emitting for a single partition?    Right now, our `checkpointUpdated` approach is subject to a race condition and would break for concurrent calls anyway...